handlers/mparticle-api/src/routers/baton.ts

```diff
@@ -0,0 +1,134 @@
+import { v4 as uuidv4 } from 'uuid';
+import type { DataSubjectRequestState } from '../../interfaces/data-subject-request-state';
+import { DataSubjectRequestStatus } from '../../interfaces/data-subject-request-state';
+import type { DataSubjectRequestSubmission } from '../../interfaces/data-subject-request-submission';
+import {
+	getStatusOfDataSubjectRequest,
+	submitDataSubjectRequest,
+} from '../apis/data-subject-requests';
+import { setUserAttributesForRightToErasureRequest } from '../apis/events';
+import { getEnv } from '../utils/config';
+
+// https://github.com/guardian/baton/blob/1037c63c9bd782aed514bf6aaa38a54dabf699eb/README.md
+interface BatonRerEventRequestBase {
+	requestType: 'RER';
+}
+
+interface BatonRerEventInitiateRequest extends BatonRerEventRequestBase {
+	action: 'initiate';
+	subjectId: string;
+	subjectEmail?: string;
+	dataProvider: 'mparticlerer';
+}
+
+interface BatonRerEventStatusRequest extends BatonRerEventRequestBase {
+	action: 'status';
+	initiationReference: string;
+}
+
+export type BatonRerEventRequest =
+	| BatonRerEventInitiateRequest
+	| BatonRerEventStatusRequest;
+
+interface BatonRerEventResponseBase {
+	requestType: 'RER';
+	status: 'pending' | 'completed' | 'failed';
+	message?: string;
+}
+
+export interface BatonRerEventInitiateResponse
+	extends BatonRerEventResponseBase {
+	action: 'initiate';
+	initiationReference: string;
+}
+
+export interface BatonRerEventStatusResponse extends BatonRerEventResponseBase {
+	action: 'status';
+}
+
+export type BatonRerEventResponse =
+	| BatonRerEventInitiateResponse
+	| BatonRerEventStatusResponse;
+
+async function handleInitiateRequest(
+	request: BatonRerEventInitiateRequest,
+): Promise<BatonRerEventInitiateResponse> {
+	const submittedTime = new Date().toISOString();
+	const environment = getEnv('STAGE') === 'PROD' ? 'production' : 'development';
+
+	/**
+	 * If you wish to remove users from audiences or from event forwarding during the waiting period,
+	 * set a user attribute and apply audience criteria and/or forwarding rules to exclude them.
+	 * https://docs.mparticle.com/guides/data-subject-requests/#erasure-request-waiting-period
+	 */
+	try {
+		await setUserAttributesForRightToErasureRequest(
+			environment,
+			request.subjectId,
+			submittedTime,
+		);
+	} catch (error) {
+		console.warn(
+			'It was not possible to set the User Attribute to remove user from audiences or from event forwarding during the waiting period.',
+			error,
+		);
+	}
+
+	// Submit
+	const dataSubjectRequestSubmission: DataSubjectRequestSubmission =
+		await submitDataSubjectRequest({
+			regulation: 'gdpr',
+			requestId: uuidv4(),
+			requestType: 'erasure',
+			submittedTime,
+			userId: request.subjectId,
+			environment,
+		});
+	return {
+		requestType: 'RER',
+		action: 'initiate',
+		status: 'pending',
+		initiationReference: dataSubjectRequestSubmission.requestId,
```

Comment:
we generate a uuid and send to both mparticle and baton in order to tie the request together later.
It looks like the ref has to be valid json
https://github.com/guardian/baton/blob/1037c63c9bd782aed514bf6aaa38a54dabf699eb/README.md#rer-initiate-call:~:text=%22initiationReference%22%3A%20%22%7BAny_Valid_Json%7D%22%2C
so when passing back to baton I think it woudl need wrapping in double quotes
```suggestion
		initiationReference: '"' + dataSubjectRequestSubmission.requestId + '"',
```

Reply:
interesting, yes you are right that it seems to use plain uuid https://github.com/guardian/support-service-lambdas/blob/03a0ce20746748759fb2adbdaec6db6dfb82e8bd/handlers/zuora-rer/src/main/scala/com/gu/zuora/rer/ZuoraRerHandler.scala#L31
In that case it would be worth making the baton docs clearer for the next person if you have a chance.

Reply:
and to confirm your interpretation - there is a specific test to ensure that it works with a plain string

https://github.com/guardian/baton/blob/9108c899c40ffb82e5ce8eb1492d71bedf1b18d3/test/models/JsonFormatsSpec.scala#L592-L604